ext/neural_snow/NeuralSnow.jl

```diff
@@ -0,0 +1,223 @@
+      in NeuralSnow.convert_model_type!() or a custom constructor for the NeuralDepthModel with the desired type.",
+        )
+    end
+end
+
+"""
+    NeuralDepthModel(FT::DataType; Δt::Union{Nothing, FT}=nothing; model::Flux.Chain, α::Union{FT, Nothing})
+An outer constructor for the `NeuralDepthModel` density parameterization for usage in a snow model.
+Can choose to use custom models via the `model` argument, or a custom exponential moving average weight for the network inputs.
+The optional Δt argument (model time step, in seconds) can be used to set the lower boundary scaling on the network (when not using a custom model)
+in accordance with the paper upon initialization, equivalent to ModelTools.settimescale!(model, Δt, dtype = FT).
+If a weight for the exponential moving average is not provided, the weight is determined as the minimum of 2Δt/86400 and 1,
+if Δt is also not provided, a default value of 1/10 is used.
+"""
+function NeuralDepthModel(
+    FT::DataType;
+    Δt = nothing,
+    model::Flux.Chain = get_znetwork(),
+    α = nothing,
+)
+    usemodel = converted_model_type(model, FT)
+    weight =
+        !isnothing(α) ? FT(α) :
+        (isnothing(Δt) ? FT(1 / 10) : FT(min(2 * Δt / 86400, 1)))
+    if !isnothing(Δt)
+        ModelTools.settimescale!(usemodel, Δt, dtype = FT) #assumes Δt is provided in seconds
+    end
+    return NeuralDepthModel{FT}(usemodel, weight)
+end
+
+#Define the additional prognostic variables needed for using this parameterization:
+ClimaLand.Snow.density_prog_vars(m::NeuralDepthModel) =
+    (:Z, :P_avg, :T_avg, :R_avg, :Qrel_avg, :u_avg)
+ClimaLand.Snow.density_prog_types(m::NeuralDepthModel{FT}) where {FT} =
+    (FT, FT, FT, FT, FT, FT)
+ClimaLand.Snow.density_prog_names(m::NeuralDepthModel) =
+    (:surface, :surface, :surface, :surface, :surface, :surface)
+
+#Extend/Define the appropriate functions needed for this parameterization:
+
+"""
+    snow_depth(m::NeuralDepthModel, Y, p, params)
+
+An extension of the `snow_depth` function to the NeuralDepthModel density parameterization, which includes the prognostic 
+depth variable and thus does not need to derive snow depth from SWE and density.
+This is sufficient to enable dynamics of the auxillary variable `ρ_snow` without extension of update_density!, and avoids
+redundant computations in the computation of runoff.
+"""
+ClimaLand.Snow.snow_depth(m::NeuralDepthModel, Y, p, params) = Y.snow.Z
+
+
+"""
+    eval_nn(vmodel, z::FT, swe::FT, P::FT, T::FT, R::FT, qrel::FT, u::FT)::FT where {FT}
+Helper function for evaluating the neural network in a pointwise manner over a `ClimaCore.Field`
+and returning the output in a broadcastable way.
+"""
+function eval_nn(
+    model,
+    z::FT,
+    swe::FT,
+    P::FT,
+    T::FT,
+    R::FT,
+    qrel::FT,
+    u::FT,
+)::FT where {FT}
+    return model([z, swe, qrel, R, u, T, P])[1]
```

Comment:
Could you explain why you need the [1] index?

Reply:
model() returns a 1-element Matrix, not the float itself. Comment has been added.